### ISSUE TITLE
Linker errors.

### ISSUE THREAD

**Issue by amir9480** [ezEngine/ezEngine#133]
Hello.
I tried to compile editor but there are several linker errors.

Platform: `Win x64`
Compiler: `VS 2019`
CMake Configuration:
![image](https://user-images.githubusercontent.com/16279288/74243137-6d6fdc00-4cf4-11ea-8373-6ba3bbdcb42e.png)

LNK2019 sample errors:
```
LNK2019	unresolved external symbol "__declspec(dllimport) struct ezStringView __cdecl BuildString(char *,unsigned int,unsigned int)" (__imp_?BuildString@@YA?AUezStringView@@PEADII@Z) referenced in function "private: void __cdecl ezFormatStringImpl<unsigned int &,unsigned int &>::ReplaceString<0>(char (* const)[64],struct ezStringView *)const " (??$ReplaceString@$0A@@?$ezFormatStringImpl@AEAIAEAI@@AEBAXQEAY0EA@DPEAUezStringView@@@Z)	FileservePlugin	E:\Cpp\ezEngine\Code\EnginePlugins\FileservePlugin\Main.obj	1
```
LNK2001 sample error:
```
LNK2001	unresolved external symbol "__declspec(dllimport) public: static class ezRTTI ezInternal::ezStaticRTTIWrapper<unsigned char>::s_RTTI" (__imp_?s_RTTI@?$ezStaticRTTIWrapper@E@ezInternal@@2VezRTTI@@A)	RendererFoundation	E:\Cpp\ezEngine\Code\Engine\RendererFoundation\Basics.obj	1	
```

Complete errors list:
[errors.log](https://github.com/ezEngine/ezEngine/files/4186062/errors.log)

**Comment by jankrassnigg** [comment]
Hey Amir, I currently can't reproduce this. Tried it with VS 2019 (latest updates) in Debug an in RelaseWithDebInfo, compiles fine for me. Do you have all the latest updates for VS 2019? Also, did you compile a specific project, or the entire solution?

Also, maybe prefer a Debug build, the VS 2019 code generation has become even less precise with a recent update, which means one of our asserts gets triggered. Will fix that sometime soon.

**Comment by amir9480** [comment]
Thanks for the response.
I have VS Version 16.4.4.
I just want to build the editor to explore and learn it.
I opened the generated solution `ezVs2019.sln` and used "Build Solution".
I also tried `ReleaseWithDebInfo` but still, errors exist.
[errors.log](https://github.com/ezEngine/ezEngine/files/4186636/errors.log)

**Comment by jankrassnigg** [comment]
In the log I don't see anything that is the root issue. But it looks like ezFoundation.lib isn't linked in, which typically means it is the one library that failed to compile somehow (in the solution that's in the "Engine" folder). You could try compiling just that and see what kind of errors you are getting. Foundation itself only depends on libraries in the "ThirdParty" folder, we've had issues in the past where MSVC sometimes stopped liking the C code in there.

It's all a bit weird, though, as I am using the same version as you. The only other ideas I have would be
1) you have something not installed, that I have, such as some specific vcredist or Windows SDK, but I would expect some kind of error in the log hinting at that
2) You have a CMake version that somehow messed up the solution? I am currently using CMake 3.14.0, but I can try latest some time.

**Comment by amir9480** [comment]
I am using CMake version 3.16.1.
So I compiled just `Foundation` solution and LNK2019 still exists.
[errors.log](https://github.com/ezEngine/ezEngine/files/4187241/errors.log)

**Comment by jankrassnigg** [comment]
Ok, I managed to reproduce your issue :)
Though I have no idea what is going wrong, and I'm not sure I have the time today to investigate. But here is what I did:

In your CMake screenshot you have two things that stand out to me. First you use the same folder for the source and the target build location. That is a bit unusual, because it pollutes your Git repository with build stuff, but it seems to work. I'd still suggest to use a different folder.

Second, you have the CMAKE_INSTALL_PREFIX set to *something*. I have literally never touched this option ever before and always left it at the default (which is C:/Program Files (x86)/ezVs2019, for me). Once I changed this to some other (none existing) folder, I am now seeing the same errors as you.

So try cleaning your CMake cache for the project, regenerate, and then leave that option at the default. That *should* fix it. And if you can find an explanation what kind of effect this setting has, do let me know :-)

**Comment by jankrassnigg** [comment]
So, actually, using the same folder for source and build output does not work. We generate cpp files for our unity builds (to speed up compilation) and those files end up between your source files. Rerunning CMake will then recursively add those cpp files to the newly generated unity cpp files, which then creates compilation errors that say "this function already has a body". Quite a mess.

So definitely do use two different folders (maybe we should add a check in CMake for that) and completely clean your git repository. For some reason those auto generated files did not show up for me in my git client, but that might have been my user specific .gitignore file.

**Comment by amir9480** [comment]
Thanks for the help.

This issue fixed in 23e80cb.
